### DISCUSSION TITLE
QField Cloud not syncing all photos from tablet when pushing changes

### DISCUSSION THREAD

**Discussion by strrings** [opengisch/QField#3328] [Q&A]
Hello :) 

My organisation is attempting to conduct a asset management survey using QField Cloud. We have a relatively large population to survey with around 9000 features in total, with point, line and polygon geometry. The survey involves collecting information on each existing feature and taking a photo. 

The field team has two members, each taking 350 - 500 photos per day of surveying. Our issue is that the majority of the photos taken each day do not sync to QField Cloud and are only able to be extracted manually by cable. For example today we had 349 photos on QField Cloud after both pushing changes, but over 450 on each tablet. The team uses the 'Push Changes' function rather than full Synchronize to avoid having to download other team members photos. 

We have attempted making a minor change and pushing changes again to no avail, with a recent attempt stalling at "215 attachments to be uploaded" with no change overnight.  

I'm wondering if this is a volume issue, or if there is something amiss with the settings? 

We're running QField on Samsung Galaxy Tab Active3 on Android 12.

Any help would be appreciated, thank you!

**Comment by suricactus** [comment]
Hello, @strrings .

Thanks for testing QFieldCloud during beta. All feedback is invaluable!

Do you turn off your screen/switch apps during push changes? It will most probably result into interuption of image synschronization.

> I'm wondering if this is a volume issue, or if there is something amiss with the settings?

There is no extra configuration, other than making your images smaller, so you can reduce the volume and speed up the synchronization.

We have multiple ideas in mind to improve the images synchronization:

- Make sure the image upload will continue if you make QField in apps background.
- Configurure images to be "push only", so you never download the images of others.
- Make images load on demand when there is internet connection.
- [...]

If you are interested in any of these getting developed faster, please drop us an email at info [at] opengis.ch .

**Reply by strrings** [reply]
Hi @suricactus 

Thank you for your advice. Just wanted to let you know we're still experiencing this issue. I can confirm that when syncing we leave the screen on and don't switch apps. We haven't been able to reliably reproduce the issue at this point, however we have noticed that there are several photos on the tablet that aren't in the attribute table, these are those that have been taken and then removed from that particular feature and re-taken. I'm wondering if the sync could be interrupted by photos whose path isn't present in the attribute table? Just a thought :) 

Thanks for your help

**Comment by Grizou79** [comment]
Hi
I had a similar problem - yesterday i recorded ~150 features (with photo). Back home, i charged the phone's battery and pushed changes in qfield. after a few moments it showed "xx pictures are uploaded in background".
Today I saw that the features were uploaded, including the path to the picture, but there are no pictures online.
i found several "QFieldPicture20221031_091658743379086804461890jpg"-files on my phone, but not the correct pictures. In Qfield they are displayed correctly when looking at the feature. So the pictures seem to be on the phone, but I cannot find them. would be hard to rename all the "QfieldPicturexyz"-files, so ist there a possibility to recover the pictures?

**Reply by strrings** [reply]
Hi Grizou79, its really frustrating isn't it! To recover the photos from the device we do the following: 
Open project > Settings > Open project folder > Select kebab menu on DCIM folder and 'Export to folder'. 
Choose a local folder on your device and export there. You can then plug your device into your computer and the pictures will be available to copy from there with the set naming convention. The export process seems to take a while so the photos won't be available instantly. 
Hope this helps!

**Comment by suricactus** [comment]
@strrings @Grizou79 Hey, just checking on you if you have updated to some of the latest QField versions. We believe the problem should be solved for pretty much all cases, as photos are now uploaded in the background. Can you confirm you experience the problem much less frequently in the last month or so?

**Comment by CACG1022** [comment]
I am currently using the latest version and I am still experiencing this problem. It is really very frustrating to have this problem with the photos, since when managing many devices for data collection, it is very complex to do the extraction via USB with each of the devices.

**Reply by ZsanettMed** [reply]
Hi!
We kindly request you to use the QFieldCloud chatbot for any support related queries. Our chatbot is designed to provide quick and efficient solutions to your problems and will open support tickets where needed. 
Thank you for your cooperation.
Best regards, 
Zsanett
